spec/unit/pdk/cli/set/config_spec.rb

```diff
@@ -0,0 +1,283 @@
+require 'spec_helper'
+require 'pdk/cli'
+
+describe 'PDK::CLI::Set::Config' do
+  describe '#run' do
+    subject(:run) { PDK::CLI::Set::Config.run(cli_opts, cli_args) }
+
+    let(:type_opt) { nil }
+    let(:as_opt) { nil }
+    let(:force_opt) { nil }
+    let(:cli_opts) do
+      {
+        type: type_opt,
+        as: as_opt,
+        force: force_opt,
+      }
+    end
+    let(:setting_name) { nil }
+    let(:setting_value) { nil }
+    let(:cli_args) { [setting_name, setting_value].compact }
+
+    let(:pdk_config) { MockEmptyConfig.new }
+
+    # Note, this class name needs to be unqiue in the ENTIRE rspec suite!
+    class MockEmptyConfig < PDK::Config
+      def user_config
+        @user_config ||= PDK::Config::Namespace.new('user') { ; }
+      end
+
+      def system_config
+        @system_config ||= PDK::Config::Namespace.new('system') { ; }
+      end
+
+      def project_config
+        @project_config ||= PDK::Config::Namespace.new('project') { ; }
+      end
+    end
+
+    before(:each) do
+      allow(PDK).to receive(:config).and_return(pdk_config)
+      allow($stdout).to receive(:puts)
+    end
+
+    RSpec.shared_examples 'a missing name error' do
+      it 'raises with missing name' do
+        expect { run }.to raise_error(PDK::CLI::ExitWithError, %r{name is required})
+      end
+    end
+
+    RSpec.shared_examples 'a missing value error' do
+      it 'raises with missing value' do
+        expect { run }.to raise_error(PDK::CLI::ExitWithError, %r{value is required})
+      end
+    end
+
+    RSpec.shared_examples 'a failed conversion error' do
+      it 'raises with failed conversion' do
+        expect { run }.to raise_error(PDK::CLI::ExitWithError, %r{error occured converting .* into a .*})
+      end
+    end
+
+    RSpec.shared_examples 'a un-settable setting error' do
+      it 'raises with failed conversion' do
+        expect { run }.to raise_error(PDK::CLI::ExitWithError, %r{can not have a value set})
+      end
+    end
+
+    RSpec.shared_examples 'a saved setting' do |setting_name, expected_value, force = nil|
+      it 'saves the setting' do
+        expect(pdk_config).to receive(:set).with(setting_name, Object, force: force).and_call_original
```

Comment:
nit: The idiomatic way to express a wildcard parameter in rspec-mocks is `anything` rather than `Object`

Reply:
Fixed.